Add docs for Fleet's Puppet module (#13837)
fleetdm/fleet: docs/Using Fleet/Puppet-module.md

# Puppet module

_Available in Fleet Premium_

Use [Fleet's Puppet module](https://forge.puppet.com/modules/fleetdm/fleetdm/readme) to automatically install custom configuration profiles on your macOS hosts based on host attributes you define in Puppet.

The module also includes functions for releasing a macOS host from await configuration and sending any custom MDM commands.

## Setup

> To setup the Puppet module, we will do the following steps:

Reviewer: there's a detail here that I think it's implicit, pointing out just in case: you need to [install](https://www.puppet.com/docs/puppet/8/modules_installing.html) Fleet's Puppet module before doing any of this.

Author: Ah, thanks Roberto! I added an "install the Puppet module" step.

We want to be explicit in the docs and assume our users are following the instructions step by step.

# Puppet module

_Available in Fleet Premium_

Use [Fleet's Puppet module](https://forge.puppet.com/modules/fleetdm/fleetdm/readme) to automatically install custom configuration profiles on your macOS hosts based on host attributes you define in Puppet.

> The module also includes functions for releasing a macOS host from await configuration and sending any custom MDM commands.

Author: It is an Apple term. On second look, Apple gives it proper noun treatment so we should as well.

I also added a link to add more context.

## Setup

> To setup the Puppet module, we will do the following steps:

Reviewer 2: ```suggestion
To set up the Puppet module, we will do the following steps:
```
(One-word "setup" is usually a noun)

> The module also includes functions for releasing a macOS host from [Await Configuration}(https://developer.apple.com/documentation/devicemanagement/release_device_from_await_configuration) and sending any custom MDM commands.

Reviewer 2: Small syntax error w/ this link:
```suggestion
The module also includes functions for releasing a macOS host from [Await Configuration](https://developer.apple.com/documentation/devicemanagement/release_device_from_await_configuration) and sending any custom MDM commands.
```

1. Install the Puppet module
2. Configure Puppet to talk to Fleet using Hiera
3. Set Fleet as a reporter

### Step 1: install the Puppet module

Install [Fleet's Puppet module](https://forge.puppet.com/modules/fleetdm/fleetdm/readme). For more instructions on how to install Puppet modules, check out the Puppet docs [here](https://www.puppet.com/docs/puppet/8/modules_installing.html).

### Step 2: configure Puppet to talk to Fleet using Heira

> 1. Create an API-only use with the GitOps role in Fleet. Instructions for creating an API-only user in Fleet are [here](./fleetctl-CLI.md#create-an-api-only-user).

Reviewer 2: ```suggestion
1. Create an API-only user with the GitOps role in Fleet. Instructions for creating an API-only user in Fleet are [here](./fleetctl-CLI.md#create-an-api-only-user).
```

2. Get the API token for your new API-only user. Learn how [here](./fleetctl-CLI.md#get-the-api-token-of-an-api-only-user).

3. Set `fleetdm::host` and `fleetdm::token` values to your Fleet server's URL and the API token respectively. Here's an example of the Hiera YAML:

```yaml
fleetdm::host: https://fleet.example.com
fleetdm::token: your-api-token 
```

Puppet docs on configuring Hiera are [here](https://www.puppet.com/docs/puppet/6/hiera_config_yaml_5.html).

> If you have staging and production Puppet environments, you can optionally set different values for each environments. This allows you to have your staging and production environment that talk to separate staging and production Fleet servers.

Reviewer 2: ```suggestion
If you have staging and production Puppet environments, you can optionally set different values for each environments. This allows you to have your staging and production environments that talk to separate staging and production Fleet servers.
```
